rastervision/core/raster_stats.py

```diff
@@ -17,20 +17,18 @@ def compute(self, raster_sources):
 
         def chip_stream(channel):
             for raster_source in raster_sources:
-                windows = raster_source.get_extent().get_windows(
-                    chip_size, stride)
-                for window in windows:
-                    chip = raster_source.get_raw_chip(window).astype(
-                        np.float32)
-                    chip = chip[:, :, channel].ravel()
-                    # Ignore NODATA values.
-                    chip[chip == 0.0] = np.nan
-                    yield chip
-
-        # Sniff the number of channels.
-        window = raster_sources[0].get_extent().get_windows(chip_size,
-                                                            stride)[0]
-        nb_channels = raster_sources[0].get_raw_chip(window).shape[2]
+                with raster_source.activate():
+                    windows = raster_source.get_extent().get_windows(
+                        chip_size, stride)
+                    for window in windows:
+                        chip = raster_source.get_raw_chip(window).astype(
+                            np.float32)
+                        chip = chip[:, :, channel].ravel()
+                        # Ignore NODATA values.
+                        chip[chip == 0.0] = np.nan
+                        yield chip
+
+        nb_channels = len(raster_sources[0].channel_order)
```

Comment:
I think we actually want to compute the stats for all channels, not just the ones selected by channel_order. Things downstream could break because of this change?

Reply:
Ah sorry I must have got confused by which was approved - I'll address these comments as a follow up.